plugins/grpc/client_closesend_interceptor.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package grpc
+
+import (
+	"github.com/apache/skywalking-go/plugins/core/operator"
+	"github.com/apache/skywalking-go/plugins/core/tracing"
+)
+
+type ClientCloseSendInterceptor struct {
+}
+
+func (h *ClientCloseSendInterceptor) BeforeInvoke(invocation operator.Invocation) error {
+	endSnapshot := tracing.GetRuntimeContextValue(END_SNAPSHOT)
```

Comment:
In grpc-go, client streaming starts with `NewStream`, but we use the `stream` to operate, for example:

```go
stream, err := NewStream()
go func() {
    for{
        req, err := stream. Recv()
    }
}()
go func() {
    for{
        err := stream. Send()
    }
}()
```

In order to make the span of `send` and `recv` be the child span of `NewStream`,we should get runtime context after `NewStream` span creation.

But even if `AsyncFinish()` is called in another goroutine, the `NewStream` span is still active in the context after the service ends. In order not to mix the next service with this service, we also need to record the context when `NewStream` ends. When the service ends, use it to restore the end state of NewStream

Reply:
Do you mean `stream.Recv()` and `stream.Send()` is decided by the user application or grpc-go?

Also, can we use `EnhancedInstance` to enhance the streaming rather than runtime context, because it will copy all context when creating a new goroutine.

Reply:
It is decided by the user application.

Do you mean I need to use `EnhancedInstance` to set and get context?

Reply:
> Do you mean I need to use EnhancedInstance to set and get context?

Yes. Also, I think it records the basic information enough(such as method name, new stream span, etc. ). And let the `NewStream` as a local span, 
`Recv` and `Send` as exit span. `Recv` and `Send` only should append to the current execution goroutine. 

Imagine the following scenario, I create streaming on a goroutine(such as initializing the system), and send/recv streaming on another goroutine. If you keep the init goroutine tracing context, it will interfere with the context of the current request. WDYT?
```
var stream protocol.Streaming
func Init(connection) {
   stream = protocol.NewStreaming(connection)
}

func HandleRequest() {
   stream.Send/Recv()
}
```